drawBot/drawBotDrawingTools.py

```diff
@@ -462,6 +463,39 @@ def pdfImage(self):
         self._drawInContext(context)
         return context.getNSPDFDocument()
 
+    def shareImage(self, format="pdf", service="airdrop", **kwargs):
+        """
+        Share the canvas to a service with a specified format.
+
+        As default the `format` is `pdf`, any suffix drawBot supports is possible.
+
+        `service` options are `airdrop`, `mail` or `message`.
+
+        .. downloadcode:: shareImage.py
+
+            # set A4 page size
+            newPage(200, 200)
+            # draw something
+            text("Foo, bar", (10, 10))
+            # share it over airdrop
+            shareImage('pdf', service="airdrop")
+        """
+        path = tempfile.mkstemp(suffix=f".{format}")[1]
+
+        context = getContextForFileExt("pdf")
```

Comment:
Btw. why does this go via the pdf context and not via the regular `saveImage()`? Would this work for anything but pdf?

Reply:
cause it needs to be `format` this is wrong of me :)

Reply:
Question still stands: why not plain `saveImage()`?

Reply:
must be `self.saveImage(..)` indeed